### PR TITLE
feat: show name and menu in header only when user is logged in

### PR DESCRIPTION
**Pull request by subhoghoshX**
This is the first of the series of PRs that improves the Sign In and Sign Up flow. Please look at #140.

This PR hides the name and menu button when the user is not logged in. Instead, it only shows a link to the Sign In page.

### CODE REVIEW

**Review by ykdojo — APPROVED** [review]
looks good, thank you!

File: src/components/Header.tsx [hunk 1]
```diff
@@ -52,89 +54,100 @@ export function Header() {
...
+													leaveFrom='transform opacity-100 scale-100'
+													leaveTo='transform opacity-0 scale-95'
+												>
+													<Menu.Items className='absolute right-0 mt-2 w-48 origin-top-right rounded-md bg-white py-1 shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none'>
+														<Menu.Item>
+															{({ active }) => (
+																<a
+																	href='#'
+																	className={classNames(
+																		active ? 'bg-gray-100' : '',
```
Comment (benny123tw) [inline src/components/Header.tsx:99]:
```suggestion
																		active && 'bg-gray-100',
```
is this one better?

**Comment by subhoghoshX** [comment]
> maybe use [clsx](https://github.com/lukeed/clsx#readme) for constructing className strings conditionally?

I'm not sure about adding another dependency for a simple case like this.

File: src/components/Header.tsx (hunk 1 above)
Comment (subhoghoshX) [reply src/components/Header.tsx:99]: In this case if `active` is true it'll return a string that is `bg-gray-100` but if it is false it'll return a boolean i.e. `false`. So a class named `false` will probably be added to the list of class names.

And as we're using TypeScript it'll straight away show an error saying something like the value of `boolean` is not assignable, if that makes sense :smile:  Thanks for the feedback.

Comment (benny123tw) [reply src/components/Header.tsx:99]: `false` will filter out by `classes.filter(Boolean).join(' ')` XD 
Thanks for the comment ❤️

Comment (subhoghoshX) [reply src/components/Header.tsx:99]: You're right. Didn't notice it before. But the TypeScript error would still be there if we use `active && 'bg-gray-100'`.

**Pull request merged**